### PR TITLE
Include query in sign in redirect

### PR DESCRIPTION
**Pull request by mrsdizzie**
Include query string in ctx.Data["CurrentURL"] so people are redirected to the right page after singing in. Fixes case of:

Visit:

 https://try.gitea.io/mrsdizzie/testcase/issues?q=&type=all&sort=&state=open&labels=7071&milestone=0&assignee=0

Sign in, then redirected to:

 https://try.gitea.io/mrsdizzie/testcase/issues

Create a new "CurrentURL" instead of updating the existing "Link" because too many places in the code assume "Link" won't have a query

### CODE REVIEW

**Comment by silverwind** [comment]
Maybe call it `OriginalURL`? Would make its intend clearer imho.

File: modules/context/context.go [hunk 1]
```diff
@@ -242,6 +242,7 @@ func Contexter() macaron.Handler {
 		}
 		ctx.Data["Language"] = ctx.Locale.Language()
 		c.Data["Link"] = ctx.Link
+		ctx.Data["SignInLink"] = setting.AppSubURL + c.Req.URL.RequestURI()
```
Comment (mrsdizzie) [inline modules/context/context.go]:
```suggestion
		ctx.Data["CurrentURL"] = setting.AppSubURL + c.Req.URL.RequestURI()
```

File: templates/base/head_navbar.tmpl [hunk 2]
```diff
@@ -131,7 +131,7 @@
 					{{svg "octicon-person" 16}} {{.i18n.Tr "register"}}
 				</a>
 			{{end}}
-			<a class="item{{if .PageIsSignIn}} active{{end}}" rel="nofollow" href="{{AppSubUrl}}/user/login?redirect_to={{.Link}}">
+			<a class="item{{if .PageIsSignIn}} active{{end}}" rel="nofollow" href="{{AppSubUrl}}/user/login?redirect_to={{.SignInLink}}">
```
Comment (mrsdizzie) [inline templates/base/head_navbar.tmpl]:
```suggestion
			<a class="item{{if .PageIsSignIn}} active{{end}}" rel="nofollow" href="{{AppSubUrl}}/user/login?redirect_to={{.CurrentURL}}">
```

**Comment by mrsdizzie** [comment]
@silverwind updated

**Review by silverwind — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**